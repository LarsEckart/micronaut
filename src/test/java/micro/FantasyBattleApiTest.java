package micro;

import org.approvaltests.JsonJacksonApprovals;
import org.junit.jupiter.api.Test;

import io.micronaut.runtime.server.EmbeddedServer;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.restassured.response.Response;
import jakarta.inject.Inject;

import static io.restassured.RestAssured.given;

@MicronautTest
class FantasyBattleApiTest {

    @Inject EmbeddedServer server;

    @Test
    void exposes_health_endpoint_due_to_micronaut_management_dependency() {
        Response response = given()
                .port(server.getPort())
                .when()
                .get("/inventory")
                .then()
                .extract()
                .response();

        JsonJacksonApprovals.verifyAsJson(response.getBody().print());
    }

}
