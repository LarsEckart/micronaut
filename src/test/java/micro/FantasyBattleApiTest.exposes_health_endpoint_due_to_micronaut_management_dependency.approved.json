"{\"equipment\":{\"rightHand\":{\"name\":\"any\",\"baseDamage\":30,\"damageModifier\":17.0},\"leftHand\":{\"name\":\"any\",\"baseDamage\":0,\"damageModifier\":0.5},\"head\":{\"name\":\"any\",\"baseDamage\":0,\"damageModifier\":0.5},\"chest\":{\"name\":\"any\",\"baseDamage\":0,\"damageModifier\":0.5},\"feet\":{\"name\":\"any\",\"baseDamage\":0,\"damageModifier\":0.5}}}"